pkg/kubelet/network/cni/cni.go

```diff
@@ -43,6 +44,8 @@ type cniNetworkPlugin struct {
 	host           network.Host
 	execer         utilexec.Interface
 	nsenterPath    string
+	mu             sync.Mutex
+	podMutexes     map[string]*sync.Mutex
```

Comment:
> So this PR should not be needed at all? Can we guarantee that network plugin calls are never in parallel? If so, maybe the fix, @dcbw, is just to document that :)

@thockin we cannot currently guarantee this, but I think we should document somewhere that we want to, and then work towards it.